cql3/restrictions/statement_restrictions.cc

```diff
@@ -1068,6 +1068,110 @@ std::vector<query::clustering_range> get_single_column_clustering_bounds(
     return ck_ranges;
 }
 
+// In old v1 indexes the token column was of type blob.
+// This causes problems because blobs are sorted differently than the bigint token values that they represent.
+// Tokens are encoded as 8 byte big endian two's complement signed integers,
+// which with blob sorting makes them ordered like this:
+// 0, 1, 2, 3, 4, 5, ..., bigint_max, bigint_min, ...., -5, -4, -3, -2, -1
+// Because of this clustering restrictions like token(p) > -4 and token(p) < 4 need to be translated
+// to two clustering ranges on the old index.
+std::vector<query::clustering_range> get_index_v1_token_range_clustering_bounds(
+        const query_options& options,
+        const column_definition* token_column,
+        const expression& token_restriction) {
+
+    // A workaround in order to make possible_lhs_values work properly.
+    // possible_lhs_values looks at the column type and uses this type's comparator.
+    // This is a problem because when using blob's comparator, -4 is greater than 4.
+    // This makes possible_lhs_values think that an expression like token(p) > -4 and token(p) < 4
+    // is impossible to fulfill.
+    // Create a fake token column with the type set to bigint, translate the restriction to use this column
+    // and use this restriction to calculate possible lhs values.
+    column_definition token_column_bigint = *token_column;
+    token_column_bigint.type = long_type;
+    expression new_token_restrictions = replace_column_def(token_restriction, &token_column_bigint);
+
+    std::variant<value_list, nonwrapping_range<managed_bytes>> values =
+        possible_lhs_values(&token_column_bigint, new_token_restrictions, options);
+
+    return std::visit(overloaded_functor {
+        [](const value_list& list) {
+            if (list.empty()) { // Impossible condition -- no rows can possibly match.
+                return std::vector<query::clustering_range>();
+            }
+
+            std::vector<query::clustering_range> ck_ranges;
+            ck_ranges.reserve(list.size());
+
+            for (auto&& value : list) {
+                ck_ranges.emplace_back(query::clustering_range::make_singular(std::vector<managed_bytes>{value}));
+            }
+
+            return ck_ranges;
+        },
+        [](const nonwrapping_interval<managed_bytes>& range) {
+            auto int64_from_be_bytes = [](const managed_bytes& int_bytes) -> int64_t {
```

Comment:
Optional: if you made this a function, it could be reused by other code some day.

Reply:
If it were supposed to be used in other places it would probably be the best to add more conversion functions - to `int32_t`, `uint32_t`, etc..
Some `convert_to_be_bytes<T>` 
I'm not sure if I should pollute global space with random functions, maybe some other more though out PR could add full conversions header